aioesphomeapi/host_resolver.py

```diff
@@ -159,6 +145,28 @@ def async_addrinfos_from_ips(ips: list[str], port: int) -> list[AddrInfo] | None
     return None
 
 
+def async_addrinfos_from_zeroconf_cache(
+    zeroconf_manager: ZeroconfManager | None, hosts: list[str], port: int
+) -> list[AddrInfo] | None:
+    """Convert a list of IPs to AddrInfos."""
+    if not zeroconf_manager or not zeroconf_manager.has_instance:
+        return None
+    aiozc = zeroconf_manager.get_async_zeroconf()
```

Comment:
needs to handle OSError and return `None`

https://github.com/esphome/aioesphomeapi/pull/1072/files#diff-4b9b26ad12930f8b24b706ca0fc82f78bd64ccdbdaf24936c3b6e780781289c4R260